pkg/startup/startup_linux.go

```diff
@@ -100,3 +103,41 @@ func ipv6Supported() bool {
 func ensureNetworkForOS(ctx context.Context, client client.Interface, nodeName string) error {
 	return nil
 }
+
+func MonitorIPAddressSubnets() {
+	autoDetectPollingInterval := DEFAULT_AUTODETECT_POLL_INTERVAL
+	if os.Getenv("AUTODETECT_POLL_INTERVAL") != "" {
+		autoDetectPollingInterval, _ = time.ParseDuration(os.Getenv("AUTODETECT_POLL_INTERVAL"))
+	}
+
+	// Add a subscription to get updated if there is any change to interface addresses.
+	addrUpdate := make(chan netlink.AddrUpdate)
+	done := make(chan struct{})
+	if err := netlink.AddrSubscribe(addrUpdate, done); err != nil {
```

Comment:
Did you confirm if this monitors for just IP addresses in the root network namespace, or will we get addr notifications for other netns as well?

It would be bad, for example, if we got notified if pod addresses coming and going :D

Reply:
It's bad :grin: 

I played around a bit with a stub program but didn't verify behavior with transient container. With `AddrSubscribe`, we get notified as soon as an container is created or destroyed. We can do better in terms of identifying new addresses only but IIU your comment correctly, it's equally bad.

I will propose the following improvement to make it acceptable --
We start with a list of available interfaces and build a map of link-indexs (with link-name which is not usable atm) and once we receive an address update we check if the update corresponds to any of the available interface. We log a new interface update but do nothing further. Only when the address update corresponds to the available interfaces, we will check for the node address change. I sort-of verified that the address update coming for a transient container doesn't match the list we started with using a stub program. Let me know what you think.

Thanks

Reply:
Did you try using `AddrSubscribeAt` and passing in the current netns via `netns.Get()`? 

https://godoc.org/github.com/vishvananda/netns#Get